### ISSUE TITLE
jax.random.poisson hangs if lam is NaN

### ISSUE THREAD

**Issue by geoff-davis** [google/jax#9719]
The following causes Python to hang:

```python
import jax
import numpy as np

key = jax.random.PRNGKey(0)
jax.random.poisson(key=key, lam=jnp.array([np.nan]), shape=(1,))
# hangs
```

My expectation is that when lam=np.nan, the output will be np.nan.

**Comment by jakevdp** [comment]
Thanks for the report - it looks like this corner case leads to a very large (but not technically infinite) loop: https://github.com/google/jax/blob/main/jax/_src/random.py#L1104

We should be able to fix this pretty quickly.

**Comment by jakevdp** [comment]
The fix is in #9721. Note that we can't return NaN in this case, because the dtype of the returned array is integer. But the fix returns `-1`, similar to other invalid values of `lam`.

**Comment by geoff-davis** [comment]
Wow, that was fast. Thanks!
